lib/graphql/stitching/shaper.rb

```diff
@@ -0,0 +1,84 @@
+# typed: false
+# frozen_string_literal: true
+
+module GraphQL
+  module Stitching
+    class Shaper
+      def initialize(supergraph:, document:)
+        @supergraph = supergraph
+        @document = document
+      end
+
+      def perform(raw_result)
+        if ! raw_result.key?("errors") || raw_result["errors"].empty?  # TODO should do null cleanup for errors too
```

Comment:
This should still run in the presence of errors. It's definitely possible to both have data results and have errors.

Reply:
That said, if there is no `data` section on the raw result – that constitutes a skip condition.

Reply:
why would we want to shape erroneous data? the shaper then needs to be robust enough for virtually any error, doesn't it?